hydromt/data_adapter/utils.py

```diff
@@ -0,0 +1,105 @@
+"""Utility functions for data adapters."""
+import logging
+import os
+from os.path import isdir, join
+from pathlib import Path
+from typing import List, Optional
+
+import numpy as np
+import pandas as pd
+import xarray as xr
+
+
+def netcdf_writer(
+    obj: xr.Dataset | xr.DataArray,
+    data_root: str | Path,
+    data_name: str,
+    variables: Optional[List[str]] = None,
+) -> str:
+    """Utiliy function for writing a xarray dataset/data array to a netcdf file.
+
+    Parameters
+    ----------
+    obj : xr.Dataset | xr.DataArray
+        Dataset.
+    data_root : str | Path
+        root to write the data to.
+    data_name : str
+        filename to write to.
+    variables : Optional[List[str]], optional
+        list of dataset variables to write, by default None
+
+    Returns
+    -------
+    fn_out: str
+        Absolute path to output file
+    """
+    dvars = [obj.name] if isinstance(obj, xr.DataArray) else obj.data_vars
+    if variables is None:
+        encoding = {k: {"zlib": True} for k in dvars}
```

Comment:
perhaps this needs to be an argument as well? or must it always be `zlib` for netcdf?

Reply:
it was already hard-coded in the geodatasetadapter as zlib. I now made it a default for this function.